README.md

```diff
@@ -93,9 +106,279 @@ src/
+
+```
+// v good
+<ThemeProvider theme={(defaultTheme) => ({
+  ...defaultTheme,
+  someThemeProperty: "#000",
+})}>
+  ...
+</ThemeProvider>
+```
+
+# Numbers
+
+## Always use Big or BigNumber for crypto amounts
+
+For all numbers that describe crypto amounts (prices, volumes, etc), store and
+manipulate them as Bigs. Floating point imprecision makes it unsuitable for
+cryptocurrency amounts.
+
+## Always manipulate numbers using Big functions
+
+Use Big's `plus`, `minus`, `div`, and `times` to perform math. Use those
+specific names, don't use their aliases.
+
+## If something needs to be a real `number`, parse it as late as possible
+
+If something like a charting library needs to use real JavaScript numbers, wait
+until the last possible second to parse the Big values. (If possible, do it
+right as you call the function that needs numbers.)
+
+## You don't need to use `toString()`
+
+Referencing a Big number will return a string, and things like Math.min/max will
+cast strings as integers.
+
+# Text
+
+## Use the right symbols
+
+Always use the correct typographic symbol for apostrophes, double-quotes,
+ellipses, etc.
+
+```js
+return <div>“It’s going to be super sweet…”</div>;
+```
+
+## Always localize text
+
+If your project involves translations, get used to always converting visible strings to text.
+
+## Never localize variables
+
+Don't localize variables. Instead, localize the string literal.
+
+```js
+// not good!
+const text = "This is some cool text!";
+return <Trans>{text}</Trans>;
+
+// do this instead
+const text = i18n.t`This is some cool text!`;
+return text;
+
+// or this
+return <Trans>This is some cool text!</Trans>;
+```
+
+## Don't break up sentences
+
+Remember that a human is going to see each translatable string separately and
+has to translate them into another language, which could have completely
+different notions of word order, sentence structure, etc.
+
+```js
+// not good!
+  <Trans>This text has</Trans>
+  {values}
+  <Trans>in the middle</Trans>
+
+// good
+  <Trans>
+    An order has been filled for {amount} {currency}
+  </Trans>
+```
+
+## Use `i18n.t` for template literals
+
+```js
+// broken (will silently not be translated)
+i18n._(`hello`);
+
+// still broken
+i18n.t(`hello`);
+
+// works
+i18n.t`hello`;
+```
+
+# Redux
+
+## In action creators, don't run `dispatch()` in a `try`
+
+If you do, then any React error caused by a change to props via
+`mapDispatchToProps` will get handled by the `catch` block and swallowed
+inappropriately.
+
+To fix, don't call `dispatch` inside a try, or inside an `actionWithStatus`,
+which wraps the action with a `try {} catch {}`.
+
+Use `setTimeout` to achieve this:
+
+```js
+async () => {
+  try {
+    const data = await fetchSomething();
+    setTimeout(() => dispatch({ type: SUCCESS, payload: data }));
+  } catch (e) {
+    dispatch({ type: FAILURE, payload: { /* ... */ }, 0);
+  }
+};
+```
+
+# Fetching
+
+## Don't modify API responses
```

Comment:
Maybe
```suggestion
## Don't modify API responses for data stored with code
```
? I think this is about when we save external data to the repo for whatever reason, but I read this a little ambiguously.